forward/forwarder.go

```diff
@@ -85,14 +85,14 @@ func (f *Forwarder) mergeDefaultOptions(opts *Options) *Options {
 
 // A Forwarder is used to forward requests to their destinations
 type Forwarder struct {
+	events.AsyncEventEmitter
```

Comment:
Not sure I agree with the API here, since it will add `Forwarder.AddListener`, `Forwarder.RemoveListener` fields. It is not obvious they are about event emission.